api/redhatopenshift/resource-manager/Microsoft.RedHatOpenshift/preview/2024-06-10-preview/openapi.json

```diff
@@ -0,0 +1,1521 @@
+        },
+        "httpsProxy": {
+          "type": "string",
+          "description": "https proxy config"
+        },
+        "noProxy": {
+          "type": "string",
+          "description": "no proxy config"
+        },
+        "trustedCa": {
+          "type": "string",
+          "description": "The trusted CA for the proxy"
+        }
+      }
+    },
+    "TokenClaimMappingsProfile": {
+      "type": "object",
+      "description": "External auth claim mappings profile",
+      "properties": {
+        "username": {
+          "$ref": "#/definitions/ClaimProfile",
+          "description": "The claim mappings username"
+        },
+        "groups": {
+          "$ref": "#/definitions/ClaimProfile",
+          "description": "The claim mappings groups"
+        }
+      },
+      "required": [
+        "username",
+        "groups"
+      ]
+    },
+    "TokenClaimValidationRuleProfile": {
+      "type": "object",
+      "description": "External auth claim validation rule",
+      "properties": {
+        "claim": {
+          "type": "string",
+          "description": "Claim"
+        },
+        "requiredValue": {
+          "type": "string",
+          "description": "Required value"
+        }
+      },
+      "required": [
+        "claim",
+        "requiredValue"
+      ]
+    },
+    "TokenIssuerProfile": {
+      "type": "object",
+      "description": "Token issuer profile",
+      "properties": {
+        "url": {
+          "type": "string",
+          "description": "The URL of the token issuer"
+        },
+        "audiences": {
+          "type": "array",
+          "description": "The audience of the token issuer",
+          "items": {
+            "type": "string"
+          }
+        },
+        "ca": {
+          "type": "string",
+          "description": "The issuer of the token"
+        }
+      },
+      "required": [
+        "url",
+        "audiences",
+        "ca"
+      ]
+    },
+    "VersionProfile": {
+      "type": "object",
+      "description": "Versions represents an OpenShift version.",
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "ID is the unique identifier of the version.",
+          "x-ms-mutability": [
+            "update",
+            "create"
+          ]
+        },
+        "channelGroup": {
+          "type": "string",
+          "description": "ChannelGroup is the name of the set to which this version belongs. Each version belongs to only a single set.",
+          "x-ms-mutability": [
+            "create"
+          ]
+        },
+        "availableUpgrades": {
+          "type": "array",
+          "description": "AvailableUpgrades is a list of version names the current version can be upgraded to.",
+          "items": {
+            "type": "string"
+          },
+          "readOnly": true
+        }
+      },
+      "required": [
+        "id",
+        "channelGroup",
+        "availableUpgrades"
```

Comment:
While reviewing this, Swagger is saying that:
> Read only properties cannot be marked as required by a schema.
Is this something that's relevant to us?

Reply:
I am honestly not sure. I see it invalid for openapi, but at the same moment the azure opanpi validator and typespec are not telling me this is an error. 

I propose, we wait until we run against current production linter when submit for ARM review. If we fail, it is easy to fix.